### PR TITLE
HIVE-22276 Break up DDLSemanticAnalyzer - extract View related analyzers

### PR DESCRIPTION
**Pull request by miklosgergely**
DDLSemanticAnalyzer is a huge class, more than 4000 lines long. The goal is to refactor it in order to have everything cut into more handleable classes under the package  org.apache.hadoop.hive.ql.exec.ddl:

- have a separate class for each analyzers
- have a package for each operation, containing an analyzer, a description, and an operation, so the amount of classes under a package is more manageable

Step #6: extract all the vire related analyzers from DDLSemanticAnalyzer, and move them under the new package.

### CODE REVIEW

File: ql/src/test/results/clientnegative/materialized_view_no_transactional_rewrite_2.q.out [hunk 1]
```diff
@@ -27,4 +27,4 @@ POSTHOOK: type: CREATE_MATERIALIZED_VIEW
 POSTHOOK: Input: default@cmv_basetable
 POSTHOOK: Output: database:default
 POSTHOOK: Output: default@cmv_mat_view
-FAILED: SemanticException Automatic rewriting for materialized view cannot be enabled if the materialized view uses non-transactional tables
+FAILED: SemanticException Automatic rewriting for materialized view cannot  be enabled if the materialized view uses non-transactional tables
```
Comment (jcamachor) [inline ql/src/test/results/clientnegative/materialized_view_no_transactional_rewrite_2.q.out]: knitpick: double spacing?

Comment (miklosgergely) [reply ql/src/test/results/clientnegative/materialized_view_no_transactional_rewrite_2.q.out]: Fixed.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rewrite/AlterMaterializedViewRewriteDesc.java [hunk 2]
```diff
@@ -27,17 +27,17 @@
...
 @Explain(displayName = "Alter Materialized View Rewrite", explainLevels = { Level.USER, Level.DEFAULT, Level.EXTENDED })
 public class AlterMaterializedViewRewriteDesc implements DDLDescWithWriteId {
-  private final String fqMaterializedViewName;
+  private final String materializedViewName;
   private final boolean rewriteEnable;
 
-  public AlterMaterializedViewRewriteDesc(String fqMaterializedViewName, boolean rewriteEnable) {
-    this.fqMaterializedViewName = fqMaterializedViewName;
+  public AlterMaterializedViewRewriteDesc(String materializedViewName, boolean rewriteEnable) {
+    this.materializedViewName = materializedViewName;
```
[the remark this answers is not in the document]
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rewrite/AlterMaterializedViewRewriteDesc.java]: Put back the fq prefix.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLSemanticAnalyzerFactory.java [hunk 3]
```diff
@@ -52,8 +55,11 @@ private DDLSemanticAnalyzerFactory() {
   private static final Map<Integer, Class<? extends BaseSemanticAnalyzer>> TYPE_TO_ANALYZER = new HashMap<>();
 
   static {
-    Set<Class<? extends BaseSemanticAnalyzer>> analyzerClasses =
+    Set<Class<? extends BaseSemanticAnalyzer>> analyzerClasses1 =
         new Reflections("org.apache.hadoop.hive.ql.ddl").getSubTypesOf(BaseSemanticAnalyzer.class);
```
[the remark this answers is not in the document]
Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLSemanticAnalyzerFactory.java]: Fixed.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLSemanticAnalyzerFactory.java [hunk 4]
```diff
@@ -29,10 +29,13 @@
 import org.apache.hadoop.hive.ql.metadata.Hive;
 import org.apache.hadoop.hive.ql.parse.ASTNode;
 import org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer;
+import org.apache.hadoop.hive.ql.parse.CalcitePlanner;
 import org.reflections.Reflections;
 
 import com.google.common.annotations.VisibleForTesting;
 
+import avro.shaded.com.google.common.collect.Sets;
```
Comment (jcamachor) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLSemanticAnalyzerFactory.java]: We should not use the shaded version.

Comment (miklosgergely) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLSemanticAnalyzerFactory.java]: Fixed.

**Pull request closed without merging**